lib/Bio/KBase/ObjectAPI/functions.pm

```diff
@@ -2585,6 +2590,104 @@ sub func_build_metagenome_metabolic_model {
+                                                              'async_version' => 'release',
+                                                            )
+                                                 );
+		my $ra = new RAST_SDK::RAST_SDKClient( $self->{'callbackURL'},
+                                                            ( 'service_version' => 'release',
+                                                              'async_version' => 'release',
+                                                            )
+                                                 );
+		my $pa = new ProkkaAnnotation::ProkkaAnnotationClient( $self->{'callbackURL'},
+                                                            ( 'service_version' => 'release',
+                                                              'async_version' => 'release',
+                                                            )
+                                                 );
+
+		my $mgap = Bio::KBase::kbaseenv::ws_client()->get_objects2({objects => [{ref => $assembly_ref}]})->{data}->[0];
+
+		#Reads ref is not in assembly object, accqurie from assembly object provenance data
+		my $readsProvRef;
+		for (my $j=0; $j< @{$mgap->{provenance}}; $j++){
+			my $readsProvList = $mgap->{provenance}->[$j]->{method_params};
+			for (my $k=0; $k< @{$readsProvList}; $k++){
+				if (defined $readsProvList->[$k]->{read_libraries}){
+					$readsProvRef = $readsProvList->[$k]->{read_libraries}->[0];
+					print &Dumper ($readsProvRef);
+				}
+			}
+		}
+
+		my $maxbinOut = $maxb->run_max_bin ({
+			assembly_ref => $assembly_ref,
+			workspace_name => $params->{workspace},
+			reads_list => [$readsProvRef],
+			binned_contig_name => 'binnedOut',
+			prob_threshold => 0.8,
+			markerset => '107',
+			min_contig_length => 1000,
+			plotmarker => 0
+
+		});
+		my $bContigs = $handler->util_get_object(Bio::KBase::utilities::buildref($maxbinOut->{binned_contig_obj_ref},$params->{workspace}));
+
+		print "Following bins will be extracted\n";
+		for (my $i =0; $i< @{$bContigs->{bins}}; $i++){
+
+			print $bContigs->{bins}->[$i]->{bid}."\n";
+		}
+
+		my $binExt = $be->extract_binned_contigs_as_assembly({
+			binned_contig_obj_ref => $maxbinOut->{binned_contig_obj_ref},
+			extracted_assemblies => '',
+			assembly_suffix => "_assembly",
+			workspace_name => $params->{workspace},
+			assembly_set_name => 'extracted_bins.AssemblySet'
+		});
+
+
+		#Annoate with Prokka
+		for (my $j=0; $j < @{$binExt->{assembly_ref_list}}; $j++){
+
+		 	my $ProkAnno = $pa->annotate({
+			output_workspace => $params->{workspace},
+			object_ref => $binExt->{assembly_ref_list}->[$j],
+			output_genome_name => $binExt->{assembly_ref_list}->[$j]."_Prokka"
+
+			});
+
+		}
+
+
+		#Annoate with RAST
+		#The API call is here, but I could not work test locally as it cannot copy the kmer ref data file through sdk call back.
+=head
+		for (my $j=0; $j < @{$binExt->{assembly_ref_list}}; $j++){
```

Comment:
So this is a case where mocking is really helpful. Basically in your tests, you override the call out the the SDK module and run some other code instead. I'm not an expert in perl mocks but I think this is the lib we've used: https://metacpan.org/pod/Mock::Quick

Alternately, you could just test this on docker01 which has the room to run RAST

Reply:
Also, it would be better to just use a flag (e.g. use_prokka) and if statements to determine which toolkit was used rather then commenting out the code.